src/typechecker.erl

```diff
@@ -2626,9 +2626,11 @@ type_check_logic_op_in(Env, ResTy, Op, P, Arg1, Arg2) when Op == 'andalso'; Op =
             {union_var_binds(VarBinds1, VarBinds2, Env#env.tenv),
              constraints:combine([Cs, Cs1, Cs2])};
         false ->
-            throw({type_error, logic_error, Op, P, ResTy})
+            {Arg2Ty, _VB, _Cs} = type_check_expr(Env#env{infer = true}, Arg2),
+            Ty = type(union, [Arg2Ty, Target]), Env#env{infer = true},
+            throw({type_error, OrigExpr, Ty, ResTy})
```

Comment:
The updated Env (with infer) isn't used for anything. It gives a compiler warning.

Reply:
Will fix in the other PR!